src/category_theory/lifting_properties.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2021 Jakob Scholbach. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Scholbach
+-/
+import category_theory.category
+import category_theory.arrow
+import category_theory.functor
+import category_theory.limits.preserves.shapes.terminal
```

Comment:
I'm not sure what the policy on reducing imports is, but you definitely don't need to import `category` and `functor`, and you can technically get away with not even importing `arrow`.

Reply:
There was recent discussion of this at https://leanprover.zulipchat.com/#narrow/stream/263328-triage/topic/PR.20.235823.3A.20chore.28*.29.20clean.20up.20file.20imports.20by.20removing.20tran.2E.2E.2E

Reply:
In the case at hand, your suggestion there would amount to using the `import category_theory.limits.preserves.shapes.terminal` only for the lemma `right_lifting_property_initial_iff`. Do you think it is worthwhile to split this file like that? I don't really care personally, and can't overlook the practical implications in the long run.

Reply:
No, I wouldn't split for that. This is a small file doing essentially one thing.

Reply:
OK, I have chosen to import `arrow` since it is somehow fundamental, and `limits.preserves.shapes.terminal` since this is strictly needed for that one lemma.